app/controllers/chat_channel_memberships_controller.rb

```diff
@@ -24,15 +24,25 @@ def chat_channel_info
     @membership = ChatChannelMembership.find(params[:id])
     authorize @membership
     @channel = @membership.chat_channel
-    data = ChatChannelDetailPresenter.new(@channel, @membership).as_json
-
-    render json: { success: true, result: data, message: "" }, success: :ok
+    invitation_link = @channel.chat_channel_invitation_links.last
+    if !invitation_link
+      invitation_link = CreateChatChannelInvitationLink.perform(@channel)
+    elsif invitation_link && (invitation_link.expiry_time < DateTime.now || invitation_link.expired?)
+      invitation_link.update(status: "expired") if invitation_link.active?
+      invitation_link = CreateChatChannelInvitationLink.perform(@channel)
+    end
+    if invitation_link&.errors&.any?
```

Comment:
Is there a chance invitation link is still nil here? It doesn't seem so, but maybe I'm wrong